<launch>
  <arg name="dataset" default="Jackal" />
  <arg name="robot_id" default="0" />
  <arg name="run_mode" default="1" /> 
  <arg name="output_path" default="$(find kimera_pgmo)/log" />
  <arg name="log" default="true" />
  <arg name="enable_sparsify" default="false"/>

  <arg name="dgrf_path"/>
  <arg name="sparse_mapping_path"/>
  <arg name="ply_path"/>
  <arg name="original_traj"/>
  <arg name="optimized_traj"/>

  <arg name="max_diff_ns" default="100000000"/>

  <node type="mesh_trajectory_deformer" name="mesh_deformer" pkg="kimera_pgmo">
    <rosparam file="$(find kimera_pgmo)/params/$(arg dataset)/parameters.yaml" />
    <param name="output_prefix" value="$(arg output_path)" />
    <param name="robot_id" value="$(arg robot_id)" />
    <param name="run_mode" value="$(arg run_mode)" />
    <param name="enable_sparsify" value="$(arg enable_sparsify)" />
    <param name="log_output" value="$(arg log)" />
    <param name="dgrf_path" value="$(arg dgrf_path)"/>
    <param name="sparse_mapping_path" value="$(arg sparse_mapping_path)"/>
    <param name="ply_path" value="$(arg ply_path)"/>
    <param name="original_traj" value="$(arg original_traj)"/>
    <param name="optimized_traj" value="$(arg optimized_traj)"/>
    <param name="max_diff_ns" value="$(arg max_diff_ns)"/>
  </node>
</launch>